subsys/net/lib/download_client/src/http_download_client.c

```diff
@@ -89,7 +89,32 @@ static int socket_sectag_set(int fd, int sec_tag)
 	return 0;
 }
 
-static int resolve_and_connect(const char *const host, int family, int sec_tag)
+static int socket_bind(int fd, const char *apn)
```

Comment:
Could we use more specific name for this operation, like `socket_apn_set`? There's a `bind` function specified in the socket API, I think that might be confusing.

Reply:
Good point, I named it `bind` because the socket option is `SO_BINDTODEVICE` but you are correct that it might be confused with `bind()` so I renamed it like you suggested.